widget/view.controller.js

```diff
@@ -0,0 +1,182 @@
+'use strict';
+(function () {
+    angular
+        .module('cybersponse')
+        .controller('fieldsOfInterest100Ctrl', fieldsOfInterest100Ctrl);
+
+    fieldsOfInterest100Ctrl.$inject = ['$scope', '$state', 'Entity', 'FormEntityService', 'Modules', 'viewTemplate', '$timeout'];
+
+    function fieldsOfInterest100Ctrl($scope, $state, Entity, FormEntityService, Modules, viewTemplate, $rootScope, $timeout) {
+        $scope.id = $state.params.id;
+        $scope.module = $state.params.module;
+        $scope.updateFieldValues = updateFieldValues;
+        $scope.notifyFieldChange = notifyFieldChange;
+        $scope.viewValueChange = viewValueChange;
+        $scope.submitField = FormEntityService.submitField;
+        $scope.$on('template:refresh', function (event, changedFields) {
+            angular.forEach(changedFields, function (field) {
+                $scope.notifyFieldChange(field.value, field);
+            });
+            init();
+        });
+
+        if (!$scope.entity) {
+            $scope.entity = FormEntityService.get();
+        }
+
+        function init() {
+            if ($scope.entity) {
+                getFields();
+                $scope.initialized = true;
+
+            }
+            var entity = new Entity($scope.module);
+            entity.get($scope.id, {
+                $relationships: true
+            }).then(function () {
+                $scope.entity = entity;
+                getFields();
+            }).finally(function () {
+                $scope.initialized = true;
+            });
+        }
+        init();
+
+        //Get all feilds which are not selected
+        function getMissingFields() {
+            var configRows = $scope.config.rows[0].columns;
+            var selectedFields = [];
+            configRows.forEach(function (column, index) {
+                column.fields.forEach(function (fields) {
+                    selectedFields.push(fields.name);
+                })
+            })
+
+            var missingFields = [];
+            var excludeArray = [];
+            $scope.config.excludeFieldsArray.forEach(function (field, index) {
+                excludeArray.push(field.name);
+            })
+            for (const key in $scope.entity.fields) {
+                if (!selectedFields.includes(key) && !excludeArray.includes(key) && $scope.entity.fields[key].type !== 'manyToMany'
```

Comment:
You will skip all M2M and O2M fields, right?

Reply:
yes